### ISSUE TITLE
New user profile is created when the old profile was deleted and option to use last used profile was selected

### ISSUE THREAD

**Issue by mblesius** [qgis/QGIS#54218]
### What is the bug or the crash?

When the option to always use the last used user profile on QGIS startup is selected, QGIS will create a new profile called _new profile_ on restart, after you deleted your most recently used profile.

### Steps to reproduce the issue

1. Open QGIS
2. Create a new profile via _Settings_ -> _User Profiles_ -> _New Profile_
3. Go to _Settings_ -> _Options_ -> _User Profiles_
4. Under _Startup Profile_ make sure _Use last closed profile_ is selected
5. Open the User profile folder, e.g. through _Settings_ -> _Open Active Profile Folder_
6. Close QGIS
7. Delete the user profile
8. Start QGIS again
9. Observe that a new user profile called _new profile_ is created

### Versions

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /></head><body>

QGIS-Version | 3.32.1-Lima | QGIS-Codeversion | 4a078ecf63
-- | -- | -- | --
Qt-Version | 5.15.3
Python-Version | 3.9.5
GDAL-Version | 3.7.1
PROJ-Version | 9.2.1
EPSG-Registraturdatenbankversion | v10.088 (2023-05-13)
GEOS-Version | 3.12.0-CAPI-1.18.0
SQLite-Version | 3.41.1
PDAL-Version | 2.5.5
PostgreSQL-Client-Version | 15.2
SpatiaLite-Version | 5.0.1
QWT-Version | 6.1.6
QScintilla2-Version | 2.13.1
BS-Version | Windows 10 Version 2009
  |   |   |  
Aktive Python-Erweiterungen
db_manager | 0.1.20
grassprovider | 2.12.99
MetaSearch | 0.3.6
processing | 2.12.99

</body></html>

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

I would find it useful if, when the option to _Use last closed profile_ is selected, QGIS would check on startup whether the last profile used still exists and then decide what to do, e.g.
1. if the last closed profile still exists -> load it, as defined in the settings
2. if it does not exists -> check if other profiles exist
3. in the case that more than one other profiles exists -> let the user choose between the existing ones
4. if only one other profile exists -> load this profile
5. if no profile exists at all -> create a new _default_ profile

**Comment by YoannQDQ** [comment]
I think "new profile" might be your default profile.

If "Use last closed profile" is selected and it does not exist, the behavior is to load the default profile (and create it if it does not exist).

I'm reluctant to display the profile selector if the user did not explicitelty asks for it, as having a dialog pop before opening QGIS migh dsturb existing workflows.

What could be done however, is as you suggested, loading the existing profile if it is the only one 

1. if the last closed profile still exists -> load it, as defined in the settings
2. if only one other profile exists -> load this profile **(NEW ⚠️)**
3. if no profile exist at all -> create a new default profile

**Comment by mblesius** [comment]
> Could you confirm whether "new profile" is indeed your default profile, and let me know what you think about the proposed 3-step resolution.

Oh wow, for whatever reason "new profile" was indeed set as the default profile on my second PC. 
I tested again and it works as expected, so I can rule out the bug I alleged.

I have given some more thought to the solution I proposed and I am no longer so comfortable with loading a profile which is not the default profile in case 4. This is mainly for the same reason you pointed out regarding the dialog. Users might also be confusing for some users when a different, non-default profile is loaded "somehow" and they cannot directly see the reason for that (perhaps they did not delete the profile themselves, thus are not aware of this).

What I would find helpful here, would be to display a message that informs the user that his last used profile was not available (?) and therefore could not be loaded and also that the default profile was used (and created) instead.
